### DISCUSSION TITLE
Problem debugging GitExtensions

### DISCUSSION THREAD

**Discussion by geerligs** [gitextensions/gitextensions#10148] [Q&A] [answered]
### Environment

- Git Extensions 33.33.33
- Build bb9b9673f4c2aa9e96410a2207712cced751f022
- Git 2.37.0.windows.1 (recommended: 2.37.1 or later)
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.7
- DPI 96dpi (no scaling)

### Issue description

After building GitExtensions I try to start a debug session but this fails with the message
![RunFailure](https://user-images.githubusercontent.com/6309628/183830122-eb097487-bf70-459e-b852-0a2dc2152b50.png)

### Steps to reproduce

1. open solution GitExtensions
2. make sure debug target is selected
3. press F5
The code starts to compile and build. After that the messagebox appears.

It is possible to run <project root>\artifacts\Debug\bin\GitExtensions\net6.0-windows\GitExtensions.exe

**Comment by geerligs** [comment]
After upgrading to VS2022 17.3 the error message is slightly changed:
![image](https://user-images.githubusercontent.com/6309628/183917914-f0d1a326-f007-457a-94e9-78d88fb9fe45.png)

**Comment by gerhardol** [comment]
Have you set "GitExtensions" as a start project?

**Comment by geerligs** [comment]
I did use GitExtensions as a start project, see attached screen shot. I also tried to use GitUI as start project but this also fails. It is not possible to use an 'empty' start project.
What should I configure/setup to start a debugging session?
![image](https://user-images.githubusercontent.com/6309628/184077715-8261224c-88b9-410e-b43d-8e87953a1e63.png)

**Comment by gerhardol** [comment, marked as answer]
This is a Visual Studio issue, setting the startup project is all I know that is required.
I found the following when googling:
https://docs.microsoft.com/en-us/answers/questions/655393/how-to-fix-34the-project-doesn39t-know-how-to-run.html

If that does not resolve your issue, you may have to ask the question in a MSVS forum.

**Reply by geerligs** [reply]
Hello Gerhard,

stupid me, I hadn't installed the .NET for desktop development workload.
Now I did (I **_had to enable_** Development Tools for .NET) and everything works fine :-)

![image](https://user-images.githubusercontent.com/6309628/184144986-f9e040e7-711b-4443-a320-a43300493e9b.png)

Thanks to the swift reply!
--Han

**Comment by RussKie** [comment]
@geerligs this looks like an issue with your local setup. 
![image](https://user-images.githubusercontent.com/4403806/184107960-1b954645-a2a2-427c-a6dc-016018827290.png)


Do you have .NET Framework installed?
Try commenting these lines and see if it makes a difference
```diff
diff --git a/GitExtensions/GitExtensions.csproj b/GitExtensions/GitExtensions.csproj
index ea34b12a5..ec89c6322 100644
--- a/GitExtensions/GitExtensions.csproj
+++ b/GitExtensions/GitExtensions.csproj
@@ -21,10 +21,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="DotnetRuntimeBootstrapper">
+    <!--<PackageReference Include="DotnetRuntimeBootstrapper">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
+    </PackageReference>-->
     <PackageReference Include="GitInfo">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
```

**Reply by geerligs** [reply]
Hello Igor,

I used your suggestion:
`    
`
Saved the csproj, close the solution and open again.

Unfortunately this doesn't make a difference :-(

--Han
